modules/byDataAnalyticsAdapter.js

```diff
@@ -0,0 +1,309 @@
+  if (!initOptions.clientId) {
+    logError('"options.clientId" should not empty!!');
+    isCorrectOption = false;
+  }
+  initOptions.isCorrectOption = isCorrectOption;
+  this.initOptions = initOptions;
+  return isCorrectOption;
+};
+
+ascAdapter.getVisitorData = function(data = {}) {
+  var ua = data.userId ? data : {};
+  var module = {
+    options: [],
+    header: [window.navigator.platform, window.navigator.userAgent, window.navigator.appVersion, window.navigator.vendor, window.opera],
+    dataos: [
+      { name: 'Windows Phone', value: 'Windows Phone', version: 'OS' },
+      { name: 'Windows', value: 'Win', version: 'NT' },
+      { name: 'iPhone', value: 'iPhone', version: 'OS' },
+      { name: 'iPad', value: 'iPad', version: 'OS' },
+      { name: 'Kindle', value: 'Silk', version: 'Silk' },
+      { name: 'Android', value: 'Android', version: 'Android' },
+      { name: 'PlayBook', value: 'PlayBook', version: 'OS' },
+      { name: 'BlackBerry', value: 'BlackBerry', version: '/' },
+      { name: 'Macintosh', value: 'Mac', version: 'OS X' },
+      { name: 'Linux', value: 'Linux', version: 'rv' },
+      { name: 'Palm', value: 'Palm', version: 'PalmOS' }
+    ],
+    databrowser: [
+      { name: 'Chrome', value: 'Chrome', version: 'Chrome' },
+      { name: 'Firefox', value: 'Firefox', version: 'Firefox' },
+      { name: 'Safari', value: 'Safari', version: 'Version' },
+      { name: 'Internet Explorer', value: 'MSIE', version: 'MSIE' },
+      { name: 'Opera', value: 'Opera', version: 'Opera' },
+      { name: 'BlackBerry', value: 'CLDC', version: 'CLDC' },
+      { name: 'Mozilla', value: 'Mozilla', version: 'Mozilla' }
+    ],
+    init: function () {
+      var agent = this.header.join(' '); var os = this.matchItem(agent, this.dataos); var browser = this.matchItem(agent, this.databrowser);
+      return { os: os, browser: browser };
+    },
+    matchItem: function (string, data) {
+      var i = 0; var j = 0; var regex; var regexv; var match; var matches; var version;
+      for (i = 0; i < data.length; i += 1) {
+        regex = new RegExp(data[i].value, 'i');
+        match = regex.test(string);
+        if (match) {
+          regexv = new RegExp(data[i].version + '[- /:;]([\\d._]+)', 'i');
+          matches = string.match(regexv);
+          version = '';
+          if (matches) { if (matches[1]) { matches = matches[1]; } }
+          if (matches) {
+            matches = matches.split(/[._]+/);
+            for (j = 0; j < matches.length; j += 1) {
+              if (j === 0) {
+                version += matches[j] + '.';
+              } else {
+                version += matches[j];
+              }
+            }
+          } else {
+            version = '0';
+          }
+          return {
+            name: data[i].name,
+            version: parseFloat(version)
+          };
+        }
+      }
+      return { name: 'unknown', version: 0 };
+    }
+  };
+  function generateUid() {
```

Comment:
@Prebid-bydata , do you have to generate uid exactly like this, would `utils.generateUUID()` work for you?

Reply:
@aleksatr hi, our backend table uid setup like this, so we have to use in this format.